### PR TITLE
Mgmolisani/feature/rich text react renderer

### PR DESCRIPTION
**Pull request by mgmolisani**
Snapshots working.
Refactored index file (its nested again since I didn't want to break it while it was working). If we want to break it back out to test, we need to create a new index file to only expose the main function export and not the helpers.
Refactored main function name to documentToReactComponents (feels better, idk)

### CODE REVIEW

File: packages/rich-text-react-renderer/src/index.tsx [hunk 1]
```diff
@@ -98,37 +90,32 @@ export function documentToReactTree(
...
-      // TODO: Figure what to return when passed an unrecognized node.
       return null;
     }
-    return renderNode[node.nodeType](node, children);
+    return renderNode[node.nodeType](node, nodeListToReactComponents(node.content, options));
   }
 }
 
-function appendKeyToValidElement(element: ReactNode, key: string | number): ReactNode {
+function appendKeyToValidElement(element: ReactNode, key: number): ReactNode {
```
Comment (connor-baer) [inline packages/rich-text-react-renderer/src/index.tsx]: What were your thoughts behind removing the `string` type? Custom keys could also be strings, right?

File: packages/rich-text-react-renderer/src/index.tsx [hunk 2]
```diff
@@ -78,15 +74,11 @@ export interface Options {
 /**
  * Serialize a Contentful Rich Text `document` to React tree
  */
-export function documentToReactTree(
+export function documentToReactComponents(
   richTextDocument: Document,
-  options: Partial<Options> = {},
+  options: Options = {},
```
Comment (connor-baer) [inline packages/rich-text-react-renderer/src/index.tsx]: Will this still work if a user passes incomplete options?

Comment (mgmolisani) [reply packages/rich-text-react-renderer/src/index.tsx]: Options already has optional types for its properties so I’m not sure the partial adds anything. Maybe we should confirm with someone else but I don’t think it has to be there.

File: packages/rich-text-react-renderer/src/index.tsx (hunk 1 above)
Comment (mgmolisani) [reply packages/rich-text-react-renderer/src/index.tsx]: That input is just a vector for the index of the child which is always a number. It is the key we inject, not a users key.

Comment (connor-baer) [reply packages/rich-text-react-renderer/src/index.tsx]: Ah, you're right 🤦‍♂️

File: packages/rich-text-react-renderer/src/index.tsx (hunk 2 above)
Comment (connor-baer) [reply packages/rich-text-react-renderer/src/index.tsx]: Alright. This was copied from `rich-text-html-renderer`, so I thought it was necessary.

**Pull request merged**